crypto/stack/stack.c

```diff
@@ -123,6 +123,22 @@ OPENSSL_STACK *OPENSSL_sk_new(OPENSSL_sk_compfunc c)
     return ret;
 }
 
+OPENSSL_STACK *OPENSSL_sk_new_reserve(int n)
+{
+    OPENSSL_STACK *st;
+
+    st = OPENSSL_sk_new_null();
+    if (st == NULL)
+        return NULL;
+
+    if (!OPENSSL_sk_reserve(st, n)) {
+        OPENSSL_sk_free(st);
+        return NULL;
+    }
```

Comment:
Side note: in other places, we've used aggregate conditions, like this:

``` C
    if ((st = OPENSSL_new_null()) == NULL
        || !OPENSSL_sk_reserve(st, n)) {
        OPENSSL_sk_free(st);
        return NULL;
    }
```

Your choice if you want to do the same

Reply:
Okay, aggregate conditions work for me